src/event.rs

```diff
@@ -53,6 +54,37 @@ impl Event {
         unsafe { ffi::gdk_event_put(self.to_glib_none().0) }
     }
 
+    /// Set the event handler.
+    /// 
+    /// The callback `func` is called for each event. If `None`, event handling
+    /// is disabled.
+    /// 
+    /// The `data` pointer is passed to `func` and `notify` when called.
+    /// 
+    /// Should another event handler be set in the future, `notify` will be
+    /// called before doing so, if not `None`.
+    pub fn set_handler<T>(
+        func: Option<&'static Fn(Event, Option<*mut T>)>,
+        data: Option<*mut T>,
+        notify: Option<&'static Fn(Option<*mut T>)>)
+    {
+        let data = data.unwrap_or(ptr::null_mut()) as *mut c_void;
```

Comment:
This "static" function need macro like `assert_initialized_main_thread!();` in first line